step-capstone/src/components/Sidebars/HandleClickedTimePoint.js

```diff
@@ -0,0 +1,75 @@
+/* Handling click into empty slot starts HERE */
+// Receives displayDate, returns the date string of displayDate
+const getDisplayDateString = (displayDate) => {
+  let year = displayDate.getFullYear().toString();
+  let month = (displayDate.getMonth() + 1 <10)? "0" + (displayDate.getMonth() + 1).toString() : displayDate.getMonth();
+  let date = (displayDate.getDate() < 10)? "0" + displayDate.getDate()   : displayDate.getDate();;
+  return year + "-" + month + "-" + date +   "T";
+};
+
+// Receives displayItems list, returns a list of empty intervals
+export const getEmptySlots = (displayItems, displayDate) => {
+  let emptySlots = [];
+  const displayDateString = getDisplayDateString(displayDate);
+  if (displayItems.length == 0) {
+    emptySlots.push({ 
+        startDate: new Date(displayDateString + "00:00:00"),
```

Comment:
In general you want to handle changes at the higher level data structure, rather than its string representation. This is because working with string representations can be pretty error prone (for example, maybe you miss a colon and your reviewers don't notice). So I would say you want to use date objects directly here and in several places below. You could do something like this:

```
startDate: new Date(displayDate.getFullYear(), displayDate.getMonth(), displayDate.getDate(), 0, 0, 0, 0)
endDate: new Date(displayDate.getFullYear(), displayDate.getMonth(), displayDate.getDate(), 23, 59, 59, 0)
```

Reply:
Good point. I will change it accordingly to your suggestion.